### ISSUE TITLE
Benchmarks vs jsonapi-rb?

### ISSUE THREAD

**Issue by NullVoxPopuli** [Netflix/fast_jsonapi#44]
Jsonapi-rb was written by @beauby, who also used to work on AMS (i did too). It's also way faster than AMS.

I saw the medium blog post, and i think more details about the benchmark woulld be good, too. 
For example, in my own benchmarks, with 200 records with 2 included relationships each,. I found that jsonapi-rb is 5 to 6 times faster than AMS. I can share the benchmarrk files with you if it'd help.

I have some doubts about the speed increase mentioned in the medium article, which is why i bring this up. 

I could do my own benchmarks, but i'm in the car right now. :)

Thanks for supporting/making-more-known the jsonapi spec. :D

**Comment by shishirmk** [comment]
Thanks for reaching out. We are putting together a longer writeup. Our benchmarks can be run very easily If you just run rspec you will see the results. More information about how the benchmarks are setup take a look at this file 

https://github.com/Netflix/fast_jsonapi/blob/master/spec/lib/object_serializer_performance_spec.rb

We had a very specific goal in mind so we got the speed increase by sacrificing some flexibility, focusing just on JSON API and assuming homogenous lists.

**Comment by ahx** [comment]
@shishirmk Thank you for sharing this and thinking about putting together a longer writeout. It would be great if you could add one or two sentences about *why* fast_jsonapi is faster than AMS. I am very interested in this sort of performance optimization, but I currently have no clue about where to start or what techniques to avoid in order to become/stay fast.

**Comment by minsikzzang** [comment]
benchmark against `jsonapi-rb` added https://github.com/Netflix/fast_jsonapi/pull/52

**Comment by sriniwasgr** [comment]
@NullVoxPopuli  @shishirmk  Here is the [link](https://github.com/Netflix/fast_jsonapi/blob/master/performance_methodology.md) to the performance doc. Hopefully it helps in answering your question.

**Comment by amitsuryavanshi** [comment]
I did some benchmarking with [https://github.com/amitsuryavanshi/json-parser-benchmark](script) and results (Iterations per second) were,
Warming up --------------------------------------
          jsonapi-rb    10.000  i/100ms
Calculating -------------------------------------
          jsonapi-rb    111.401  (±20.6%) i/s -      1.580k in  15.058330s


Warming up --------------------------------------
        fastjson-api   118.000  i/100ms
Calculating -------------------------------------
        fastjson-api      1.270k (± 9.6%) i/s -     18.880k in  15.014813s

Can someone confirm these results ? Let me know if I am doing anything wrong here. Thanks.

**Comment by shishirmk** [comment]
@amitsuryavanshi 
Here is the benchmark i see on my laptop based on #52 by @minsikzzang 

```
Serialize to Ruby Hash 1000 records
Serializer      Records    Time
AMS serializer  1000       376.51 ms
jsonapi-rb serializer 1000       42.34 ms
Fast serializer 1000       9.8 ms

Serialize to Ruby Hash 250 records
Serializer      Records    Time
AMS serializer  250        89.47 ms
jsonapi-rb serializer 250        8.54 ms
Fast serializer 250        2.09 ms
```

**Comment by beauby** [comment]
I would expect `fast_jsonapi` to be faster than `jsonapi-rb` as it is more specialized, but a fair comparison would make `jsonapi-rb` use `Oj` as well for JSON serialization.

**Comment by NullVoxPopuli** [comment]
@beauby I have some benchmarks here with different scenarios if you're curious: https://github.com/Netflix/fast_jsonapi/issues/81

Right now the biggest difference is relationship performance. (in favor of jsonapi-rb)

**Comment by guilleiguaran** [comment]
Switching all the serializers to `FastJsonapi::MultiToJson.to_json`:

```
Serialize to JSON string 10000 records
Serializer             Records    Time       Speed Up
Fast serializer        10000      144.85 ms
AMS serializer         10000      3413.27 ms 23.56x ✘
jsonapi-rb serializer  10000      406.75 ms  2.81x ✔
jsonapi-serializers    10000      356.9 ms   2.46x ✔

Serialize to Ruby Hash 10000 records
Serializer             Records    Time       Speed Up
Fast serializer        10000      115.26 ms
AMS serializer         10000      3358.11 ms 29.14x ✔
jsonapi-rb serializer  10000      399.27 ms  3.46x ✔
jsonapi-serializers    10000      303.01 ms  2.63x ✔
```

**Comment by NullVoxPopuli** [comment]
@guilleiguaran do you have code for those benchmarks? they look fishy

**Comment by guilleiguaran** [comment]
@NullVoxPopuli those are the benchmarks shipped with `fast_jsonapi`, see #96 for my changes.

**Comment by NullVoxPopuli** [comment]
ah ok. cool. @guilleiguaran, there are more in-depth benchmarks here, if you want to play with those https://github.com/Netflix/fast_jsonapi/issues/81

**Comment by shishirmk** [comment]
Released as a part of version 1.1.0
